### PR TITLE
CC-16934: Add Glue support in Unzer ECO module

### PR DESCRIPTION
**Pull request by michbeck**
- Developer(s): @volodymyr-bezuhlyi, @michbeck

- Ticket: https://spryker.atlassian.net/browse/CC-16934

- Release Group: https://release.spryker.com/release-groups/view/4251

- PR Overview: https://release.spryker.com/release/pull-request-eco/Unzer/29

- merge: squash

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Unzer             | minor                 |  UnzerApi                    |

-----------------------------------------

##### Change log

Improvements

- Introduced `Item.canceledAmount` transfer field.
- Introduced `Expense.canceledAmount` transfer field.
- Introduced `Expense.sumPriceToPayAggregation` transfer field.
- Introduced `Refund.amount` transfer field.
- Introduced `Refund.expenses` transfer field.
- Introduced `UnzerApiGetWebhookResponse` transfer.
- Introduced `UnzerApiDeleteWebhookResponse` transfer.
- Introduced `UnzerApiRefundResponse` transfer.
- Introduced `UnzerApiMarketplaceRefundResponse` transfer.
- Introduced `UnzerApiMessage` transfer.
- Adjusted `UnzerFacade::executePostSaveHook()` so it adds expenses only if amountPerUnit is specified and greater than 0.
- Impacted `UnzerCheckoutPostSavePlugin` with facade changes.
- Adjusted `UnzerFacade::performPreSaveOrderStack()` so it adds existing or new UnzerPaymentResource to UnzerPayment information.
- Impacted `UnzerCheckoutPreSaveOrderPlugin` with facade changes.
- Adjusted `UnzerFacade::executeRefundOmsCommand()` so it adds related order id to refund information.
- Impacted `UnzerRefundCommandByOrderPlugin` with facade changes.
- Increased `UnzerApi` version dependency.

### CODE REVIEW

File: src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+use Generated\Shared\Transfer\RestCheckoutRequestAttributesTransfer;
+use Spryker\Zed\CheckoutRestApiExtension\Dependency\Plugin\CheckoutDataExpanderPluginInterface;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+
+/**
+ * @method \SprykerEco\Zed\Unzer\Business\UnzerFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\Unzer\UnzerConfig getConfig()
+ * @method \SprykerEco\Zed\Unzer\Communication\UnzerCommunicationFactory getFactory()
+ */
+class UnzerCheckoutDataExpanderPlugin extends AbstractPlugin implements CheckoutDataExpanderPluginInterface
```
Comment (DmitryLymarenko) [inline src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php]: this plugin needs to be moved to the UnzerRestApi module

**Review by DmitryLymarenko — CHANGES_REQUESTED** [review]

File: src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php (hunk 1 above)
Comment (michbeck) [reply src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php]: Fixed.

**Review by DmitryLymarenko — COMMENTED** [review]
not all of the changes are described at changelogs
at least `UnzerFacade::performPreSaveOrderStack()` is touched
also, some distributor business model has cahnges

**Review by DmitryLymarenko — APPROVED** [review]

File: composer.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "license": "MIT",
     "require": {
         "php": ">=7.4",
-        "spryker-eco/unzer-api": "^1.0.0",
+        "spryker-eco/unzer-api": "dev-feature/cc-16934/dev-unzer-glue-support",
```
Comment (abitskil) [inline composer.json]: * Don't forget to change the version to "^1.1.0", it will not happen automatically.

Comment (michbeck) [reply composer.json]: Yep, will be done before release. Thanks for short reminder here.

**Review by kkicheglovspryker — APPROVED** [review]

**Review by abitskil — APPROVED** [review]

File: composer.json (hunk 2 above)
Comment (michbeck) [reply composer.json]: Fixed.

**Pull request merged**